nestjs-migration/src/post/post.controller.spec.ts

```diff
@@ -0,0 +1,215 @@
+				mockPostId
+			);
+			const result = await postController.handlePostCreate(
+				mockCreatePostDto,
+				mockReq
+			);
+
+			expect(postService.createPost).toHaveBeenCalledWith(
+				{...mockCreatePostDto, author_id: mockUserId}
+			);
+			expect(result).toEqual({
+				postId: mockPostId,
+				message: "게시글 생성 success",
+			});
+		});
+
+		it("게시물 생성 중 문제 발생 시 에러를 반환한다.", async () => {
+			const invalidPostDto = {
+				title: "hello",
+				content: "",
+				doFilter: true,
+			};
+
+			const error = ServerError.badRequest(
+				POST_ERROR_MESSAGES.REQUIRE_CONTENT
+			);
+			jest.spyOn(postService, "createPost").mockRejectedValue(error);
+
+      const result = await postController.handlePostCreate(invalidPostDto, mockReq);
+
+			expect(result).toEqual({
+        message: `${error.name}: ${error.message}`,
+      });
+	});
+
+	describe("GET /post", () => {
+		const mockReadPostsQueryDto = {};
+
+		it("전체 게시물 조회 성공", async () => {
+			const mockTotalPosts = 5;
+			const mockPostHeaders = [];
+
+			jest.spyOn(mockPostService, "findPostHeaders").mockResolvedValue(
+				mockPostHeaders
+			);
+			jest.spyOn(mockPostService, "findPostTotal").mockResolvedValue(
+				mockTotalPosts
+			);
+
+			expect(
+				await postController.handlePostsRead(
+					mockReadPostsQueryDto,
+					mockReq
+				)
+			).toEqual({
+				total: mockTotalPosts,
+				postHeaders: mockPostHeaders,
+			});
+		});
+
+		it("전체 게시물 조회 실패 시 에러를 반환한다.", async () => {
+			const error = ServerError.badRequest("잘못된 입력입니다");
+
+			jest.spyOn(mockPostService, "findPostHeaders").mockRejectedValue(
+				error
+			);
+			jest.spyOn(mockPostService, "findPostTotal").mockRejectedValue(
+				error
+			);
+
+			await expect(
+				postController.handlePostsRead(mockReadPostsQueryDto, mockReq)
+			).rejects.toThrow(error);
+		});
+	});
+	describe("GET /post/:post_id", () => {
+		const mockPostId = 1;
+
+		it("개별 게시물 조회 성공", async () => {
+			const mockEachPost = [];
+			jest.spyOn(mockPostService, "findPost").mockResolvedValue(
+				mockEachPost
+			);
+
+			expect(
+				await postController.handlePostRead(mockPostId, mockReq)
+			).toEqual({
+				post: [],
+			});
+		});
+		it("개별 게시물 조회 실패 시 에러를 반환한다.", async () => {
+			const error = ServerError.badRequest("잘못된 입력입니다");
+			jest.spyOn(mockPostService, "findPost").mockRejectedValue(error);
+
+			await expect(
+				postController.handlePostRead(mockPostId, mockReq)
+			).rejects.toThrow(error);
+		});
+	});
+
+	describe("PATCH /post/:post_id", () => {
+		const mockPostId = 1;
+		const updateBodyDto = {
+			title: "title",
+			content: "content",
+			doFilter: true,
+		};
+		it("게시물 수정 성공", async () => {
+			mockPostService.updatePost.mockResolvedValue([]);
+
+      const result = await postController.handlePostUpdate(
+        mockPostId,
+        updateBodyDto,
+        mockReq
+      );
+
+      expect(postService.updatePost).toHaveBeenCalledWith(mockPostId, {
+        ...updateBodyDto,
+        author_id: mockUserId,
```

Comment:
이렇게 수정하니 test 원할하게 통과됩니다.
```suggestion
        authorId: mockUserId,
```